### PR TITLE
MINIFICPP-1504: followup review changes

### PR DESCRIPTION
**Pull request by martinzink**
Renamed CPU to Cpu
Removed static_errors
Replaced c style casts to static_cast
Replaced implicit narrowing
Added type aliases

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/src/utils/OsUtils.cpp [hunk 1]
```diff
@@ -168,6 +168,11 @@ std::string OsUtils::userIdToUsername(const std::string &uid) {
   return name;
 }
 
+[[deprecated("Unsupported platform")]]
```
[the remark this answers is not in the document]
Comment (martinzink) [reply libminifi/src/utils/OsUtils.cpp:171]: In that case I am not sure if we have a good way to warn the user in a non-fatal way.
Should we just give back -1 silently?

[the remark this answers is not in the document]
Comment (szaszm) [reply libminifi/src/utils/OsUtils.cpp:171]: I would go for `#warning`. Non-MSVC compilers tend to support this GCC extension, and MSVC is handled anyway.

[the remark this answers is not in the document]
Comment (martinzink) [reply libminifi/src/utils/OsUtils.cpp]: thanks, good idea changed it

**Review by szaszm — APPROVED** [review]
Thanks for following up on the feedback despite the unfortunate timing on the other PR.

**Review by lordgamez — APPROVED** [review]
LGTM, CI failures seem to be related to [MINIFICPP-1482](https://issues.apache.org/jira/browse/MINIFICPP-1482) and [MINIFICPP-1500](https://issues.apache.org/jira/browse/MINIFICPP-1500)

**Review by adamdebreceni — APPROVED** [review]

**Comment by szaszm** [comment]
I messed up the commit msg, but meant to close this with 4fc6f49d

**Pull request closed without merging**